src/main/java/uk/gov/hmcts/reform/wataskmonitor/services/jobs/termination/TerminationJobService.java

```diff
@@ -43,8 +44,13 @@ public void terminateTasks(String serviceAuthorizationToken) {
             .filter(t -> t.getDeleteReason().equals("cancelled"))
             .collect(Collectors.toList());
 
+        List<HistoricCamundaTask> deletedTasks = tasks.stream()
+            .filter(t -> t.getDeleteReason().equals("deleted"))
+            .collect(Collectors.toList());
+
         terminateAllTasksWithReason(serviceAuthorizationToken, completedTasks, COMPLETED);
         terminateAllTasksWithReason(serviceAuthorizationToken, cancelledTasks, CANCELLED);
+        terminateAllTasksWithReason(serviceAuthorizationToken, deletedTasks, DELETED);
```

Comment:
Just pondering this.  If Camunda add another reason ... there's no reason why they can't since it doesn't break their API... we could get another problem.

So there could be some refactoring here to do where we can be more generic i.e. just try and terminate all of them?  Would have to think that through a bit with the Task Management logic too.

Reply:
Discussion taken offline a new ticket has been raised to address future  error scenarios